paimon-core/src/main/java/org/apache/paimon/utils/SegmentsCache.java

```diff
@@ -37,8 +37,13 @@ public class SegmentsCache<T> {
     private final Cache<T, Segments> cache;
     private final MemorySize maxMemorySize;
     private final long maxElementSize;
+    private final boolean cacheStatsEnabled;
```

Comment:
May this field cacheStatsEnabled could be defined in ObjectCache instead of in SegmentsCache.
What do you think?